pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -246,6 +263,54 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	return nil
 }
 
+// cleanupErroredSnapshots cleans any volumesnapshot CRs that are in Error state. It doesn't remove them
+// from the volumesnapshotschedule items list though.
+func (s *SnapshotScheduleController) cleanupErroredSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) (err error) {
```

Comment:
I didn't review the original CR so trying to understand this better about the intent here.
We delete the snapshots but not remove from the schedule list because we rely on the prune to clean it up?
If yes then prune can delete the error snapshots too right?

Reply:
We don't want to wait till prune takes care of deleting it as prune will work based on `retain` value and till that time the retries on the errored snapshots will keep happening with old cloudsnap id. So this fix will make sure the volumesnapshot gets deleted but its info remains in schedule till pruning removes it as per its own logic.